caikit/core/data_model/dataobject.py

```diff
@@ -89,6 +91,7 @@
 def dataobject(
     schema: _SCHEMA_DEF_TYPE,
     package: str = CAIKIT_DATA_MODEL,
+    optional_property_names: Optional[Set[str]] = None,
```

Comment:
Writing that out, though, it does bring up the awkwardness of `enum` schemas which would never make sense to have `optional_property_names` with and _may_ themselves have some additional "special" things like non-sequential numbering or aliases.

Reply:
Actually, why not just omit this special arg and say that if a user wants to _explicitly_ set optional fields (without the `Optional[T]` typing), they use the `jtd` format like so?

```py
@dataobject({"properties": {...}, "optionalProperties": {...}}
```

Reply:
I could be okay with saying that `Optional[T]`s get put into `optionalProperties` iff you are using the super basic shorthand that does not already have the field inside of `properties` or `optionalProperties`, and otherwise we'll just keep the fields where they are?

Reply:
Hm, that's a good point. There are pitfalls everywhere I suppose